linkerd/app/integration/src/tests/transparency.rs

```diff
@@ -843,9 +843,7 @@ macro_rules! http1_tests {
 
             let srv = server::http1()
                 .route_fn("/", |req| {
-                    let has_body_header = req.headers().contains_key("transfer-encoding")
-                        || req.headers().contains_key("content-length");
```

Comment:
I have no idea why this test--which only mentions `transfer-encoding` in its description also checks for `content-length`...  Now that we're properly preserving the size hint, we can set the `content-length` header to `0` if we know it's empty. This seems like a strictly good thing.